lisp/ess-custom.el

```diff
@@ -2108,27 +2113,26 @@ for help files.  The default value is nil for other systems."
 
 ;;;;; names for communication using MS-Windows 9x/NT ddeclient mechanism
 
-(defcustom inferior-ess-ddeclient nil
+(defcustom inferior-ess-ddeclient "Initial"
```

Comment:
I am back home.

M-x S-initiate from ESS-5.12 works correctly on Windows 10 with S-Plus 8.2
GUI.
M-x S-initiate from ESS-15.03-1 doesn't work on Windows 10 with S-Plus 8.2
GUI.


The major difference between the two versions of ess-sp6w-d.el is the name
changes of variables to drop the 6.  My plan is to revert the older
8-Nov-2010 version of the file into the current ESS and then make it work.

Once it works again, I recommend that it be frozen.  No new wonderful
style features should be incorporated into ess-sp6w-d.el

I barely use S-Plus anymore, and I rarely use Windows except for
testing compatability--needed because much of my university is windows
based.
I personally switched to Macintosh four years ago.  As one of you
commented, S-Plus itself is old---but it still works.  Therefore ESS should
continue to still work with it.

The major response that I think ESS should have to TIBCO is to
automatically detect when TERR is installed and generate the function
M-x TERR.  This would be done at the same time that the various versions
of R are detected and the M-x R-* functions are generated.

The MS DDE technology is the communications technology used by M-x S  on
Windows.  It is a one-way
communication.  C-c C-n and related keystrokes send lines from the
*ESS* buffers to the independent S-Plus GUI process.  Nothing comes
back.

M-x Sqpe on Windows works in the ESS-standard manner of running a
command-line
interpreter in an *S* inferior buffer.  Sqpe on Windows is the S-Plus
equivalent
of Rterm on Windows.


On Sun, Apr 3, 2016 at 12:04 PM, RICHARD M. HEIBERGER <rmh@temple.edu>
wrote:

> i will look at this later in the week.  my prior is that dde should be
> retained as is, with no changes, so that S-Plus gui will continue to work
> on windows.
> similarly ess-sp6w.el should be retained.
>
> when i invented the term "Initial" i had areason.  please don't change it
> until i recall the situation.
>
> Sent from my iPhone
>
> On Apr 3, 2016, at 09:25, Lionel Henry <notifications@github.com> wrote:
>
> In lisp/ess-custom.el
> <https://github.com/emacs-ess/ESS/pull/309#discussion_r58310258>:
>
> > @@ -2108,27 +2113,26 @@ for help files.  The default value is nil for other systems."
> >
> >  ;;;;; names for communication using MS-Windows 9x/NT ddeclient mechanism
> >
> > -(defcustom inferior-ess-ddeclient nil
> > +(defcustom inferior-ess-ddeclient "Initial"
>
> Why "Initial". Looks like a wart.
>
> yes, someone initialised that variable to "Initial" in ess-site.el... I
> should have changed the code to deal with nil rather than "Initial" but I
> think I was starting to give up at that point ;) I'll change it to nil.
>
> I think we can drop support for DDE. I doubt anyone uses that anymore.
>
> It's needed for S-plus on Windows. I'm not against dropping support, but
> let's see if @rmheiberger <https://github.com/rmheiberger> manages to run
> the latest ESS with these environments.
>
> No-one tests is anyways.
>
> Yes, we really need basic tests for all the inferiors officially
> supported, and some people on the team in charge of running them before
> each release. I think we should drop support for all the systems we are not
> able to test.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/emacs-ess/ESS/pull/309/files/603f35f77e2039f5fea8c52a643347b70f77c4be#r58310258>
>
>